diesel/src/mysql/types/mod.rs

```diff
@@ -8,36 +8,88 @@ use std::io::Write;
 use byteorder::WriteBytesExt;
 
 use deserialize::{self, FromSql};
-use mysql::Mysql;
+use mysql::{Mysql, MysqlType};
 use serialize::{self, IsNull, Output, ToSql};
-use sql_types;
+use sql_types::*;
 
-impl ToSql<sql_types::Tinyint, Mysql> for i8 {
+impl ToSql<Tinyint, Mysql> for i8 {
     fn to_sql<W: Write>(&self, out: &mut Output<W, Mysql>) -> serialize::Result {
         out.write_i8(*self).map(|_| IsNull::No).map_err(Into::into)
     }
 }
 
-impl FromSql<sql_types::Tinyint, Mysql> for i8 {
+impl FromSql<Tinyint, Mysql> for i8 {
     fn from_sql(bytes: Option<&[u8]>) -> deserialize::Result<Self> {
         let bytes = not_none!(bytes);
         Ok(bytes[0] as i8)
     }
 }
 
-impl ToSql<sql_types::Bool, Mysql> for bool {
+/// Represents the MySQL unsigned type.
+#[derive(Debug, Clone, Copy, Default, SqlType, QueryId)]
+pub struct Unsigned<ST>(ST);
+
+impl ToSql<Unsigned<SmallInt>, Mysql> for u16 {
+    fn to_sql<W: Write>(&self, out: &mut Output<W, Mysql>) -> serialize::Result {
+        ToSql::<SmallInt, Mysql>::to_sql(&(*self as i16), out)
```

Comment:
Just to be sure. It's okay here to convert this to a signed integer? 
This does not use some kind of undefined behavior in case of self > 128? 
I assume that the cast does not change the binary representation and only that matters when it is written to the database? (Because of that, maybe it would be clearer to do the writing directly here?)

(This applies to all `ToSql`/`FromSql` impls below.)

Reply:
No, this does not invoke any UB, nor does it change the representation in memory.